apps/hash-frontend/src/components/hooks/use-agent-runner.ts

```diff
@@ -24,20 +27,25 @@ export const useAgentRunner = <T extends AgentType["Agent"]>(
   });
 
   const callAgentRunnerCallback = useCallback(
-    async (input: Extract<AgentType, { Agent: T }>["Input"]) => {
-      return (
-        await callAgentRunnerFn({
-          variables: {
-            payload: {
-              Agent: agent,
-              Input: input,
-            },
-          },
-        })
-      ).data?.callAgentRunner.Output;
+    async (input: Parameters<CallAgentRunnerCallback<T>>[0]) => {
+      const { data, errors } = await callAgentRunnerFn({
+        variables: {
+          payload: { Agent: agent, Input: input },
+        },
+      });
+
+      const output = data?.callAgentRunner.Output as Extract<
+        AgentType,
+        { Agent: T }
+      >["Output"];
+
```

Comment:
The cast here is a leftover from a previous attempt at fixing the TS error thrown. You're right the cast should account for `undefined`, but because we are casting the method at the end of the hook we don't actually need to cast here at all. Have removed it in https://github.com/hashintel/hash/pull/2486/commits/7b414b1743d0b6730b8a6245d698b1a1ee831066